react_11/11-targil/guessingGame/src/main.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { useState } from 'react';
+
+const App = () => {
+  
+  const [target] = useState(Math.floor(Math.random() * 1000));
+
+  const [input,setInput] = useState(0);
+
+  function handleInputChange(e) {
+    setInput(e.target.value);
+  }
+
+  return (
+    <div>
+      <h2>Guessing game</h2>
+      <label>Guess the number:<input type="number" value={input} onChange={handleInputChange}/></label>
+      <br/><GuessResult guess={input} target={target}/>
+    </div>
+  )
+};
+
+function GuessResult(props) {
```

Comment:
There is no need for braking down this into 2 components. You can do everything in 1.
Never try to prepare things for the future but try to write the most simple code for what you need at the moment.